File: purchase-tester-java/src/main/java/com/revenuecat/purchasetester/java/Navigator.java
package com.revenuecat.purchasetester.java;

import android.content.Context;
import android.content.Intent;

public class Navigator {

    static void startCatsActivity(Context context, boolean clearBackStack) {
        Intent intent = new Intent(context, CatsActivity.class);
        if (clearBackStack) {
            intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_NEW_TASK);
        }
        context.startActivity(intent);
    }

    static void startUpsellActivity(Context context, boolean clearBackStack) {
        Intent intent = new Intent(context, UpsellActivity.class);
        if (clearBackStack) {
            intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_NEW_TASK);
        }
        context.startActivity(intent);
    }

}

File: purchases-sample/src/main/java/com/revenuecat/sample/navigator.kt
package com.revenuecat.sample

import android.content.Context
import android.content.Intent

fun Context.startCatsActivity(clearBackStack: Boolean = false) {
    Intent(this, CatsActivity::class.java).also {
        if (clearBackStack) {
            it.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK)
        }
    }.let {
        startActivity(it)
    }
}

fun Context.startUpsellActivity(clearBackStack: Boolean = false) {
    Intent(this, UpsellActivity::class.java).also {
        if (clearBackStack) {
            it.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK)
        }
    }.let {
        startActivity(it)
    }
}
